libs/core/runtime_local/src/runtime_local.cpp

```diff
@@ -57,6 +57,27 @@
 #include <thread>
 #include <utility>
 
+namespace hpx{ namespace detail {
+    void log_all_runtime_threads()
+    {
+        auto rt = hpx::get_runtime_ptr();
+        if (rt)
+        {
+            rt->get_thread_manager().enumerate_threads(
+                [](hpx::threads::thread_id_type id) -> bool {
+                    hpx::threads::thread_data* td = get_thread_id_data(id);
+                    auto sched = td->get_scheduler_base();
+                    LTM_(debug).format(
+                        "Logging all runtime threads: pool({}), scheduler({}),"
+                        "thread({}), description({}), state({})",
+                        sched->get_parent_pool(), sched, id,
+                        td->get_description(), td->get_state().state());
+                    return true;
+                });
+        }
+    }
+}};    // namespace hpx::detail
```

Comment:
Should we wrap this in a `#if defined(HPX_HAVE_LOGGING)` as well?

Reply:
I'd also suggest to wrap all of the body into a `try {} catch (...) {}` as this is being invoked in a signal handler and we can't be sure that all of the runtime is in a consistent and valid state at this point any more.

Reply:
Thanks!